src/modules/rangeproof/main_impl.h

```diff
@@ -304,4 +304,98 @@ int secp256k1_rangeproof_sign(const secp256k1_context* ctx, unsigned char *proof
      proof, plen, min_value, &commitp, blind, nonce, exp, min_bits, value, message, msg_len, extra_commit, extra_commit_len, &genp);
 }
 
+int secp256k1_rangeproof_verify_value(const secp256k1_context* ctx, const unsigned char* proof, size_t plen, uint64_t value, const secp256k1_pedersen_commitment* commit, const secp256k1_generator* gen) {
+    secp256k1_ge commitp;
+    secp256k1_ge genp;
+    secp256k1_gej tmpj;
+    secp256k1_gej xj;
+    secp256k1_ge rp;
+    secp256k1_scalar es;
+    secp256k1_scalar ss;
+    secp256k1_sha256 sha2;
+    unsigned char tmpch[33];
+    unsigned char pp_comm[32];
+    size_t offset;
+    size_t sz;
+    int overflow;
+
+    VERIFY_CHECK(ctx != NULL);
+    ARG_CHECK(proof != NULL);
+    ARG_CHECK(commit != NULL);
+    ARG_CHECK(gen != NULL);
+
+    if (plen != 73 && plen != 65) {
+        return 0;
+    }
+    /* 0x80 must be unset for any rangeproof; 0x40 indicates "has nonzero range"
+     * so must also be unset for single-value proofs */
+    if ((proof[0] & 0xC0) != 0x00) {
+        return 0;
+    }
+
+    secp256k1_pedersen_commitment_load(&commitp, commit);
+    secp256k1_generator_load(&genp, gen);
+    /* Verify that value in the header is what we expect; 0x20 is "has nonzero min-value" */
+    if ((proof[0] & 0x20) == 0x00) {
+        if (value != 0) {
+            return 0;
+        }
+        offset = 1;
+    } else {
+        int i;
+        uint64_t claimed = 0;
+        for (i = 0; i < 8; i++) {
+            claimed = (claimed << 8) | proof[1 + i];
+        }
```

Comment:
nit: you can loop here with `for (offset = 1; offset < 9; offset++)`, then get rid of the `int i` declaration, the `+1` in the indexing, and the assignment `offset = 9` below.

Reply:
Thank you! The target here is very memory-constrained devices so any reduction in the number of variables is good.

Reply:
The compiler would probably make this change for you, I suspect. But you could also initialize offset to 1 where its declared, remove the assignment to 1 in the if branch above, and use `while (offset < 9) { ..; offset++; }` as your loop here.

You could probably reorder the stack vars to align them and use less stack space too.

Reply:
Updated the loop but didn't reorder any variables (*that* I think even the most limited compiler would do)